hive-watcher/hive-watcher.py

```diff
@@ -128,38 +157,74 @@ def allowed_op_id(operation_id) -> bool:
         return False
 
 
-def output(post, quiet=False, use_test_node=False) -> int:
-    """Prints out the post and extracts the custom_json"""
+def write_post_to_csv(post,filepath_data):
+    fieldnames = [
+        'timestamp_seen','timestamp_post',
+        'id','type','trx_id','trx_num','block_num',
+        'required_auths','required_posting_auths','json'
+    ]
+    # build a new custom dictionary from the post
+    post_row={
+        'timestamp_seen':repr(time.time()),
+        'timestamp_post':repr(post.get("timestamp").timestamp()),
+        'id':str(post.get("id")),
+        'type':str(post.get("type")),
+        'trx_id':str(post.get("trx_id")),
+        'trx_num':repr(post.get("trx_num")),
+        'block_num':repr(post.get("block_num")),
+        'required_auths':repr(post.get("required_auths")),
+        'required_posting_auths':repr(post.get("required_posting_auths")),
+        'json':str(json.dumps(json.loads(post.get("json")), indent=4))
+    }
+    # csv writer, see: https://docs.python.org/3/library/csv.html
+    if os.path.isfile(filepath_data) :
+        with open(filepath_data,'a', newline='') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            writer.writerow(post_row)
+            csvfile.close
+    else :
+        with open(filepath_data,'w', newline='') as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerow(post_row)
+            csvfile.close
+
+def output(post, quiet=False, use_test_node=False, write_csv=False,post_type="data_undefined") -> int:
+    if write_csv:
+        write_post_to_csv(post,post_type + ".csv")
 
+    """Prints out the post and extracts the custom_json"""
     data = json.loads(post.get("json"))
     if quiet:
         if data.get("num_urls"):
             return data.get("num_urls")
         else:
             return 1
 
-    data["required_posting_auths"] = post.get("required_posting_auths")
-    data["trx_id"] = post.get("trx_id")
-    data["timestamp"] = post.get("timestamp")
-
     count = 0
-    if use_test_node:
-        data["test_node"] = True
+    try: # if any error in looking for podcast url info just return count=0
```

Comment:
This is a bit lazy, these errors are expected if  --include-nonpodping is included, should just test if post.item('url') or post.item('urls') exist, also would need to decide if counting nonpodping items makes sense.